modules/azure/storage_account_public/main.tf

```
@@ -116,3 +116,119 @@ resource "azurerm_storage_management_policy" "storage_management_policy" {
     }
   }
 }
+
+data "azurerm_monitor_diagnostic_categories" "blob" {
+  count       = var.loganalytics_diagnostic_setting.workspace_id == null || var.loganalytics_diagnostic_setting.blob == null ? 0 : 1
+  resource_id = "${azurerm_storage_account.storage_account.id}/blobServices/default/"
+}
+
+data "azurerm_monitor_diagnostic_categories" "queue" {
+  count       = var.loganalytics_diagnostic_setting.workspace_id == null || var.loganalytics_diagnostic_setting.queue == null ? 0 : 1
+  resource_id = "${azurerm_storage_account.storage_account.id}/queueServices/default/"
+}
+
+data "azurerm_monitor_diagnostic_categories" "table" {
+  count       = var.loganalytics_diagnostic_setting.workspace_id == null || var.loganalytics_diagnostic_setting.table == null ? 0 : 1
+  resource_id = "${azurerm_storage_account.storage_account.id}/tableServices/default/"
+}
+
+data "azurerm_monitor_diagnostic_categories" "file" {
+  count       = var.loganalytics_diagnostic_setting.workspace_id == null || var.loganalytics_diagnostic_setting.file == null ? 0 : 1
+  resource_id = "${azurerm_storage_account.storage_account.id}/tableServices/default/"
```

I have not checked if this is correct, but should it not be like this?
```suggestion
  resource_id = "${azurerm_storage_account.storage_account.id}/fileServices/default/"
```

---

You are Right! My bad